var/spack/repos/builtin/packages/krb5/package.py

```diff
@@ -30,7 +30,7 @@ class Krb5(AutotoolsPackage):
     version("1.16.1", sha256="214ffe394e3ad0c730564074ec44f1da119159d94281bbec541dc29168d21117")
 
     depends_on("diffutils", type="build")
-    depends_on("bison", type="build")
+    depends_on("bison%gcc", type="build")
```

Comment:
This is not how to fix it, better to put a conflict on bison for gcc.

Reply:
Is the problem that bison is mis-compiled by oneapi compilers?

Reply:
please read this message:
https://github.com/spack/spack/issues/37172#issuecomment-1561872297
(the problem is that bison compiled with oneapi has an unexpected behaviour, according to @joaquintorres

Reply:
but yes, then maybe we could conflict bison for oneapi instead.

Reply:
Would be good to report that issue upstream. Do we have a small example that shows that bison compiled with gcc vs oneapi produces different output?

Reply:
Sorry, I don't have any experience with bison. Probably  @joaquintorres would be the one to ask

Reply:
Ok yeah, we could add a `conflicts("%oneapi")` on bison, a comment explaining it doesn't work in practice, and live with that. But *ideally* we get something actionable for the Intel compiler folks :)

Reply:
ok I did that.

Reply:
Sorry, I just read this. I also don't have any experience with bison. I basically followed the logs to the compilation issues that stemmed from the incorrect code generation for the corresponding C file. @Thomas-Ulrich  Did you report this upstream to Intel?  I wouldn't mind doing this myself so we can have a proper fix instead of a workaround. AFAICT the only channel to do this seems to be through the Intel Forums, though I'm unclear if @haampie  got some more info about possible optimizations causing flags causing this on https://github.com/spack/spack/issues/37172#issuecomment-1810457784

Reply:
No I did not report anything, sorry about that.